runtime/src/qio/sys.c

```diff
@@ -1532,10 +1533,10 @@ sys_sockaddr_t sys_getaddrinfo_addr(sys_addrinfo_ptr_t a) {
 }
 sys_addrinfo_ptr_t sys_getaddrinfo_next(sys_addrinfo_ptr_t a) {return a->ai_next;}
 
-void sys_freeaddr_info(sys_addrinfo_ptr_t *p)
+void sys_freeaddrinfo(sys_addrinfo_ptr_t p)
 {
-  freeaddrinfo(*p);
-  *p = NULL;
+  freeaddrinfo(p);
+  p = NULL;
```

Comment:
I think the intent here was to set the freed pointer to NULL, but I see it didn't match the `extern proc` and that is why you made this change. So, also remove the `p = NULL` line here, because the value of `p` won't be communicated back to the call site in C.